packages/sdk/src/index.ts

```diff
@@ -1,16 +1,24 @@
 import _ from 'lodash';
+import BN from 'bn.js';
 import type { SubmittableExtrinsic } from '@polkadot/api/types';
 import { u8aToHex } from '@polkadot/util';
 import type { HexString } from '@polkadot/util/types';
 import type { KeyringPair } from '@polkadot/keyring/types';
 import { ChainAdapter, OakAdapter, TaskSchedulerChainAdapter, SendExtrinsicResult, XcmInstructionNetworkType } from '@oak-network/adapter';
+import { Weight } from '@oak-network/sdk-types';
 
 interface ScheduleXcmpTaskWithPayThroughSoverignAccountFlowParams {
   oakAdapter: OakAdapter;
   destinationChainAdapter: ChainAdapter;
   taskPayloadExtrinsic: SubmittableExtrinsic<'promise'>;
   schedule: any;
   keyringPair: KeyringPair;
+  options?: {
+    taskPayloadEncodedCallWeight?: Weight;
+    taskPayloadOverallWeight?: Weight;
+    taskPayloadExecutionFeeAmount?: BN;
+    taskPayloadInstructionCount?: number,
```

Comment:
No, the issue about XCM does not come with the encoded call, so there’s no need to specify `taskPayloadEncodedCallWeight` here, which should always be calculated. All the option variables are calculated with the 3 basic values so these are the ones to be over-written.

Also the variables are not properly named. Please use the below variables in option,

xcmOptions:{
instructionCount,
instructionWeight,
feePerSecond, 
}

> **encodedCallWeight** has nothing to do with xcm, so it should always be calculated by rpc calls.
> WEIGHT_REF_TIME_PER_SECOND is a constant and doesn’t need to be over-written for now since every chain is using the same value.

### instructionCount and instructionWeight
These foundamental values determine the overall XCM weight 

### feePerSecond
This determines the conversion from XCM weight to fee. It’s named unitsPerSecond in your code, but `feePerSecond` is more meaningful.